### DISCUSSION TITLE
RA6M5 USB HS Enumeration stops

### DISCUSSION THREAD

**Discussion by mcuenthusiast** [hathach/tinyusb#2215] [Q&A]
Hi All

I am trying latest code for developing usb host msc, and during enumeration I found that device gets disconnected before attempting for further enumeration steps.

<img width="593" alt="image" src="https://github.com/hathach/tinyusb/assets/132841642/d8ead2c4-74f5-40e0-ae53-fbf3f1a81d2c">

I am also have doubt on below API.
<img width="433" alt="image" src="https://github.com/hathach/tinyusb/assets/132841642/6116945e-4384-4745-887c-cbb2af970ae3">

This APIs checks for the Host connect status using attach interrupt status flag, but this flag will be cleared when we enter into the interrupt.

So **rusb->INTSTS1_b.ATTCH** might not be true always.

<img width="656" alt="image" src="https://github.com/hathach/tinyusb/assets/132841642/225a0e33-4670-4bbf-99a7-b042b6473e1c">

@hathach did you faced any issue due to this while testing usb hs on ra6m5?

I commented that specific code for further process of enumeration so, getting below error as per analyzer. 

<img width="960" alt="image" src="https://github.com/hathach/tinyusb/assets/132841642/7724a170-ce16-498b-888b-abb537398316">

**Comment by mcuenthusiast** [comment]
@hathach could you please let me know if you have faced any issue while integrating USB Highspeed on RA6M5 Host?

Is RA6M5 tested for USB HS Host MSC?

I am successfully able to run USB device HS on EK-RA6M5

**Reply by hathach** [reply]
I don't have any issue testing with ek-ra6m5 host on either highspeed and fullspeed last time I tested. Make sure you have the jumper setting correctly.

**Reply by mcuenthusiast** [reply]
Thanks for the response @hathach , I have verified jumper settings are correct. I have been using EK-RA6M5 for USB FS and HS with renesas example code and both ports were working with renesas example code.

**Reply by hathach** [reply]
So I guess it is all good right ?

**Reply by mcuenthusiast** [reply]
No, I mean to say USBHS port hardware connections are okay but I am still facing the issue mentioned above which using EK-R6M5 in host HS mode, Device HS is working fine.

**Reply by hathach** [reply]
have you tried the stock example with gcc ? it works for me

**Reply by mcuenthusiast** [reply]
let me check, I am using e2studio(uses gcc compiler) generated project and BSP, and integrated tinyUSB in project.

I will update you.

@hathach  which IDE do you use for programming EK-RA6M5?

**Reply by mcuenthusiast** [reply]
Hi @hathach , I have build the stock example for GCC and also flashed it using Jflash.

could you please let me know how to use embedded CLI interface for testing msc?

which tool I need or any other Hardware connections I need to do?

I am using windows PC

**Comment by mcuenthusiast** [comment]
Hi @hathach 

I tried to use stock example with ra6m5-ek with gcc for host high speed and full speed both.

I am unable to see any logs I have compiled the code using "**make BOARD=ra6m5_ek LOG=2 LOGGER=rtt all**"

but when I see the data in analyzer I am able to see data for USB FS for host but I am seeing nothing for Highspeed in analyzer.

could you please let me know what changes or command and tools I might need to use for seeing logs?

I tried to use RTT viewer but not able to see logs with that also.

With USB device on high speed and full speed I don't see any issues.

**Comment by mcuenthusiast** [comment]
@hathach Thanks for the support and response till now.

As I understand the stock example for ra6m5_ek high speed host uing gcc is working at your end.

But some how I am missing some thing while testing and building stock example (msc_file_explorer).

Below is the issue I am facing.

-  I have taken the latest tinyUSB pull and compiled using below steps
1. cd examples/host/msc_file_explorer
2. make BOARD=ra6m5_ek get-deps
3. make BOARD=ra6m5_ek all
4. make BOARD=ra6m5_ek flash-jlink

After following all above steps I am unable to see data on usb analyzer for highspeed port, instead I see the data from full speed port (PORT = 0)

After debugging I saw that even after I compile code for PORT=1 I am getting full speed data only.

So to over come this issue I have to change the tusb_config.h, and defined below macros explicitly.
#define BOARD_TUH_RHPORT      1
#define BOARD_TUH_MAX_SPEED   OPT_MODE_HIGH_SPEED

After Enabling above macros I am able to get few packets, facing the same issue I mentioned in actual post.

@hathach It will be great if you could help me to find possible issue or If I am missing any configs?

I am not able to see logs through RTT viewer(**make BOARD=ra6m5_ek LOG=2 LOGGER=rtt all**) also so it is very difficult to debug.

Any help will be highly appreciated!!!

Thanks in Advance.
